ComponentKit/Components/CKAsyncImageComponent.mm

```diff
@@ -44,31 +44,36 @@ + (instancetype)newWithURL:(NSURL *)url
   }
   CKViewComponentAttributeValueMap attributes(passedAttributes);
   attributes.insert({
-    {@selector(setSpecifier:), [[CKNetworkImageSpecifier alloc] initWithURL:url
-                                                               defaultImage:options.defaultImage
-                                                            imageDownloader:imageDownloader
-                                                                  scenePath:scenePath
-                                                                   cropRect:cropRect]},
+    {@selector(setSpecifier:), [[CKAsyncImageSpecifier alloc] initWithIdentifier:identifier
+                                                                    defaultImage:options.defaultImage
+                                                                 imageDownloader:imageDownloader
+                                                                       scenePath:scenePath
+                                                                        cropRect:cropRect]},
 
   });
   return [super newWithView:{
-    {[CKNetworkImageComponentView class], @selector(didEnterReusePool), @selector(willLeaveReusePool)},
+    {[CKAsyncImageComponentView class], @selector(didEnterReusePool), @selector(willLeaveReusePool)},
     std::move(attributes)
   } size:size];
 }
 
 @end
 
-@implementation CKNetworkImageSpecifier
+@implementation CKAsyncImageSpecifier
 
-- (instancetype)initWithURL:(NSURL *)url
-               defaultImage:(UIImage *)defaultImage
-            imageDownloader:(id<CKNetworkImageDownloading>)imageDownloader
-                  scenePath:(id)scenePath
-                   cropRect:(CGRect)cropRect
+- (instancetype)initWithIdentifier:(id)identifier
+                      defaultImage:(UIImage *)defaultImage
+                   imageDownloader:(id<CKAsyncImageRetrieval>)imageDownloader
+                         scenePath:(id)scenePath
+                          cropRect:(CGRect)cropRect
 {
   if (self = [super init]) {
-    _url = [url copy];
+    if ([identifier respondsToSelector:@selector(copyWithZone:)]) {
```

Comment:
This is the only actual change to the functioning of the code. Previously, it was always appropriate to use `NSCopying` since the param was an `NSURL`. With my change, we only copy if it is supported by the `identifier` object.